apps/prairielearn/src/pages/studentInstanceQuestionExam/studentInstanceQuestionExam.js

```diff
@@ -225,16 +226,30 @@ router.get('/variant/:variant_id/submission/:submission_id', function (req, res,
 
 router.get('/', function (req, res, next) {
   if (res.locals.assessment.type !== 'Exam') return next();
-  const variant_id = null;
-  question.getAndRenderVariant(variant_id, null, res.locals, function (err) {
-    if (ERR(err, next)) return;
-    logPageView(req, res, (err) => {
+
+  async.series(
+    [
+      (callback) => {
+        const variant_id = null;
+        question.getAndRenderVariant(variant_id, null, res.locals, (err) => {
+          if (ERR(err, callback)) return;
+          callback(null);
+        });
+      },
+      (callback) => {
+        logPageView(req, res, (err) => {
+          if (ERR(err, callback)) return;
+          callback(null);
+        });
+      },
+      async () => await setQuestionCopyTargets(res),
+    ],
+    (err) => {
       if (ERR(err, next)) return;
       question.setRendererHeader(res);
-      setQuestionCopyTargets(res);
       res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-    });
-  });
+    },
+  );
```

Comment:
Maybe we could convert this all to async/await instead of introducing `async.series()`?

Reply:
That would require adding async versions of the other functions in play here. I was trying to make the smallest amount of changes in this PR as possible; I'd prefer to save this kind of refactoring for another PR.

Reply:
I had thought we'd just `promisify` the functions that weren't converted yet but I'm also completely fine to leave this to a later PR if you prefer.